codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamFrame.java

```diff
@@ -50,11 +45,11 @@ public boolean equals(Object o) {
             return false;
         }
         Http2StreamFrame other = (Http2StreamFrame) o;
-        return streamId == other.streamId();
+        return stream == other.stream() || (stream != null && stream.equals(other.stream()));
     }
 
     @Override
     public int hashCode() {
-        return streamId;
+        return stream.hashCode();
```

Comment:
yeah only for testing. Let me think about what to do with it. Maybe we should just remove it all together  and act on the default implementation. WDYT ?